app/services/service_listeners/attachment_updater.rb

```diff
@@ -0,0 +1,67 @@
+module ServiceListeners
+  class AttachmentUpdater
+    def initialize(attachable)
+      @attachable = attachable
+    end
+
+    def update!
+      # if an attachment references an attachable, the attachable may
+      # not reference the attachment: for example, deleted attachments
+      # reference their attachable, but the attachable does not
+      # reference them.
+      @attachable.attachables.each do |attachable|
```

Comment:
If you add an `attr_reader :attachable` just below the `private` you can access this without the `@`.